libyul/optimiser/Metrics.h

```diff
@@ -29,41 +29,94 @@ namespace solidity::yul
 struct Dialect;
 struct EVMDialect;
 
+/**
+ * Weights to be assigned to specific yul statements and expressions by a metric.
+*/
+struct CodeWeights
+{
+	// Statements
+	size_t expressionStatementCost;
```

Comment:
What about assigning the default weights here as default values?

Reply:
I think that having a named constant for the defaults is more readable when you want to have defaults for function parameters.

This:
``` c++
void foo(CodeWeights const& _weights = DefaultWeights):
```
makes the intent clearer than this:
``` c++
void foo(CodeWeights const& _weights = CodeWeights{}):
```
With the second one you might think it will be zeroed.

Should I change it?

Reply:
Since zero weights are non-sensical, I would actually assume that `CodeWeights{}` produces some "sane" weight set :)

Reply:
I rarely use defaults other than `nullptr` or `0` so I would probably expect something like `CodeWeights{}` to be zeros. Zeros could be treated as a special value for when weights are not needed or something. You could just as well have several, named sets of sensible defaults.

Anyway, I'll change it. To be honest I actually stated out with the solution you're suggesting and later "improved" it because I thought it was messy :)

Reply:
Done.